plugins/in_kubernetes_events/kubernetes_events.c

```diff
@@ -550,14 +584,35 @@ static int process_events(struct k8s_events *ctx, char *in_data, size_t in_size,
     return ret;
 }
 
-static struct flb_http_client *make_event_api_request(struct k8s_events *ctx,
+static struct flb_http_client *make_event_watch_api_request(struct k8s_events *ctx,
+                                                      struct flb_connection *u_conn,                                                      
+                                                      uint64_t max_resource_version) {
+    flb_sds_t url;
+    struct flb_http_client *c;
+
+    if (ctx->namespace == NULL) {
+        url = flb_sds_create(K8S_EVENTS_KUBE_API_URI);
+    } else {
+        url = flb_sds_create_size(strlen(K8S_EVENTS_KUBE_NAMESPACE_API_URI) + 
+                                  strlen(ctx->namespace));
+        flb_sds_printf(&url, K8S_EVENTS_KUBE_NAMESPACE_API_URI, ctx->namespace);
+    }
+
+    flb_sds_printf(&url, "?watch=1&resourceVersion=%llu", max_resource_version);
+    flb_plg_info(ctx->ins, "Requesting %s", url);
+    c = flb_http_client(u_conn, FLB_HTTP_GET, url,
```

Comment:
validate `c` is not NULL

Reply:
this is done on line 865 after the return from this function